modules/lib/lib-content/src/main/resources/lib/xp/content.ts

```diff
@@ -1594,3 +1617,92 @@ export function resetInheritance(params: ResetInheritanceParams): void {
 
     bean.execute();
 }
+
+export interface UpdateMediaParams {
+    key: string
+    name: string
+    data: ByteSource
+    artist?: string
+    caption?: string
+    copyright?: string
+    focalX?: number
+    focalY?: number
+    mimeType?: string
+    tags?: string
+}
+
+interface ModifyMediaHandler {
+    setKey(value: string): void,
+
+    setName(value: string): void,
+
+    setData(value: ByteSource): void,
+
+    setFocalX(value: number): void,
+
+    setFocalY(value: number): void,
+
+    setArtist(value?: string | null): void,
+
+    setCaption(value?: string | null): void,
+
+    setCopyright(value?: string | null): void,
+
+    setMimeType(value?: string | null): void,
+
+    setTags(value?: string | null): void,
+
+    execute<Data, Type extends string>(): Content<Data, Type>,
+}
+
+/** This function modifies a media content.
+ *
+ * @example-ref examples/content/modifyMedia.js
+ *
+ * @param {object} params JSON with the parameters.
+ * @param {string} params.key Path or id to the content.
+ * @param {string} params.name Name to the content.
+ * @param {function} params.data Data (as stream) to use.
+ * @param {string} [params.mimeType] Mime-type of the data.
+ * @param {string} [params.artist] Artist to the content.
+ * @param {string} [params.caption] Caption to the content.
+ * @param {string} [params.copyright] Copyright to the content.
+ * @param {string} [params.tags] Tags to the content.
+ * @param {number} [params.focalX=0.5] Focal point for X axis (if it's an image).
+ * @param {number} [params.focalY=0.5] Focal point for Y axis (if it's an image).
+ *
+ * @returns {object} Modified content as JSON.
+ */
+export function modifyMedia<Data = Record<string, unknown>, Type extends string = string>(params: UpdateMediaParams): Content<Data, Type> | null {
+    checkRequired(params, 'data');
+    checkRequiredString(params, 'key');
+    checkRequiredString(params, 'name');
+    checkOptionalString(params, 'artist');
+    checkOptionalString(params, 'caption');
+    checkOptionalString(params, 'copyright');
+    checkOptionalString(params, 'mimeType');
+    checkOptionalString(params, 'tags');
+    checkOptionalNumber(params, 'focalX');
+    checkOptionalNumber(params, 'focalY');
+
+    const bean = __.newBean<ModifyMediaHandler>('com.enonic.xp.lib.content.ModifyMediaHandler');
+
+    bean.setKey(params.key);
+    bean.setName(params.name);
+    bean.setData(params.data);
+
+    bean.setArtist(__.nullOrValue(params.artist));
+    bean.setCaption(__.nullOrValue(params.caption));
+    bean.setCopyright(__.nullOrValue(params.copyright));
+    bean.setMimeType(__.nullOrValue(params.mimeType));
+    bean.setTags(__.nullOrValue(params.tags));
+
+    if (params.focalX) {
```

Comment:
Something I might have overlooked:

Can focalX or focalY be 0... aka setting focalpoint to upperleft corner of an image...

Because 0 is falsy https://developer.mozilla.org/en-US/docs/Glossary/Falsy

if (params.focalX) -> if (isSet(params.focalX))